mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java

```diff
@@ -129,21 +128,20 @@ public void save(Mailbox mailbox) throws MailboxException {
 
         CassandraId cassandraId = retrieveId(cassandraMailbox);
         cassandraMailbox.setMailboxId(cassandraId);
-        if (mailbox.getName().length() >= MAX_SIZE_FOR_MAILBOX_NAME) {
-            throw new TooLongMailboxNameException("too long mailbox name");
-        }
         try {
-            boolean applied = mailboxDAO.retrieveMailbox(cassandraId)
-                    .thenCompose(optional -> optional
-                            .map(storedMailbox -> mailboxPathDAO.delete(storedMailbox.generateAssociatedPath()))
-                            .orElse(CompletableFuture.completedFuture(null)))
-                    .thenCompose(any -> mailboxPathDAO.save(mailbox.generateAssociatedPath(), cassandraId))
+            boolean applied = mailboxPathDAO.save(mailbox.generateAssociatedPath(), cassandraId)
```

Comment:
Please run SMTP test for cassandra, I guess you are breaking renames

Step 1 : retrieve old path (it's not attached to the mailbox, see rename in StoreMailboxManager)
Step 2 : Try insert on mailboxPath table for new entry
Step 3 : Then update Mailbox table and remove old entry in mailboxPath table